pkg/reconciler/tenancy/clusterworkspace/clusterworkspace_reconcile_scheduling.go

```diff
@@ -87,12 +87,32 @@ func (r *schedulingReconciler) reconcile(ctx context.Context, workspace *tenancy
 			}
 
 			if len(shards) == 0 {
-				// find a shard for this workspace, randomly
+				// no specific shard was required,
+				// we are going to schedule the given ws onto the root shard
+				// this step is temporary until working with multi-shard env works
+				// until then we need to assign ws to the root shard otherwise all e2e test will break
 				var err error
 				shards, err = r.listShards(selector)
 				if err != nil {
 					return reconcileStatusStopAndRequeue, err
 				}
+
+				if len(shards) > 0 {
+					// trim the list to contain only the "root" shard so that we always schedule onto it
+					for _, shard := range shards {
+						if shard.Name == "root" {
+							shards = []*tenancyv1alpha1.ClusterWorkspaceShard{shard}
+							break
+						}
+					}
+					if len(shards) == 0 {
```

Comment:
this case surprises me. Do we need it?

Reply:
I think so, in case there are no shards we fall back to:

https://github.com/kcp-dev/kcp/blob/74a71ccccd2d40d0221c8d17a22495f0b54cf05f/pkg/reconciler/tenancy/clusterworkspace/clusterworkspace_reconcile_scheduling.go#L152

Reply:
does this need a change after our Slack discussion?